### DISCUSSION TITLE
private plugin reference

### DISCUSSION THREAD

**Discussion by developer-leon** [outlandishideas/wpackagist#162] [Q&A]
This may be more a composer question, but here it goes: 

since some paid wp plugins are not listed on the wp site they are not mirrored on wpackagist. So for those plugins, how could I handle the installation for which i already have the code (downloaded on purchase) but want to make them privately available for my project under composer?

I've tried to add them to a private repo and reference them on composer as a "svc" repository, but not quite sure how to set the values for the composer.json file. This is what I'm trying:

On my package, i added a composer.json with the following:

``` json
{
  "name": "my-wp-plugins/cool-paid-plugin",
  "type": "wordpress-muplugin",
  "version": "1.0.1",
  "require": {
    "composer/installers": "~1.0"
  }
}
```

on my main wp composer.json i added the following:

``` json
...
    {
      "type": "vcs",
      "url": "git@bitbucket.org:my-user/wp-plugin-cool-paid-plugin.git"
    }
...
```

Would that be enough? Any help or hint pointing me to the right direction will be much appreciated!

Thanks.

**Comment by tamlyn** [comment]
Does this help https://getcomposer.org/doc/05-repositories.md#package-2? See also #129.

**Comment by vercotux** [comment]
There's a composer plugin I've made that lets you include plugins/themes directly from Envato and WPMUDEV. Not perfect, but a step forward: https://github.com/MaeCompany/composer-wordpress

**Comment by positonic** [comment]
This solution will only work for people using [Wordplate](https://github.com/wordplate/wordplate), which uses wPackagist. With Wordplate, you can use [Wordplate auto-loader](https://github.com/wordplate/autoloader), and do the following:

I am using a reference to my files as the repo, but you could equally swap this out for another repository type....

```
"require": {
    "private/repo" : "dev-master"
}
"repositories": [
    {
            "type": "path",
            "url": "jamespfarrell/private-plugin"
     }
]
```


Then in the private plugin folder I have my composer.json:

```
{
  "name": "jamespfarrell/private-plugin",
  "minimum-stability" : "dev",
  "type" : "wordpress-plugin"
}
```

**Comment by tyrann0us** [comment]
I know I'm late to the party, but _SatisPress_ is what you are looking for: https://github.com/cedaro/satispress/.

It exposes installed WordPress plugins and themes as Composer packages.

**Reply by NoelLH** [reply]
Nice, thanks for adding!
